### PR TITLE
MMDP-382 - Implement new db schema 

### PR DESCRIPTION
**Pull request by fgkinus**
**What does this PR do?**
- Implement the recommended database schema for Stakeholders directory and the associated tables

**Description of Task to be completed?**
In order to allow for easy data migration, this PR implements application models to be a replica of the database schema as currently is.
- add tables and relationships
- add data integrity enforcement rules and set up validation
- add endpoints for stakeholders and returnee services
- add keystone updates to initialise the associated tables with values during server initialisation

**How should this be manually tested?**

```
git clone 
git checkout 
yarn
yarn run: server
```
==> open postman and try out the following endpoints
- To add a new stakeholder
`POST /v1/stakeholders-directory`
```
{
	"organisationName":"organisation2",
	"organisationTypeId":"5c80f368112ff1861a57145a",
	"registrationStatusId":"5c80f368112ff1861a57145e",
	"impactTypeID":"5c80f368112ff1861a571459",
	"founder":"Founder",
	"phoneNumber":"12345678",
	"email":"email@email.com",
	"website":"http://www.organisation.com",
	"description":"desc",
	"staffStrengthRangeId":"5c80f368112ff1861a571463",
	"edoStateOperationStartYear":"1994",
	"yearOfCacREG":"1995",
	"cacRcNumber":"2acE74894",
	"volunteersCount":"5",
	"localManagerName":"manager name",
	"localManagerEmail":"manager@email.com",
	"localManagerMobile":"123456789",
	"partnerWithGovernment":true,
	"challenges":"none",
	"assistanceRequired":"none",
	"locality":"Local"
}
```

- to search by the name of the stakeholder
 `GET  /v1/stakeholders-directory?organisationName=<value>`
- `PUT /v1/stakeholders-directory/<id:stakeholder_id>`
 use the schema described above to update
- to delete a stakeholder 
`DELETE  /v1/stakeholders-directory/<id:stakeholder_id>`
- to list beneficiaries
`GET v1/stakeholders-directory/<id:stakeholder_id>/beneficiaries`
- to add beneficiaries :
 `POST /v1/stakeholders-directory/<id:stakeholder_id>/beneficiaries`
```
{
    "serviceName": "Service name9",
    "sourceOfFundingId": "5c80f368112ff1861a571460",
    "averageNumberOfMaleBeneficiaries": 10,
    "averageNumberOfFemaleBeneficiaries":20,
    "averageNumberOfBeneficiary": 30,
    "amountInvestedRange": "5c80f367112ff1861a571452",
    "targetAudienceId": "5c80f368112ff1861a57146a",
    "duration": 50,
    "note": "some words",
    "beneficiaryTypeId": "5c80f367112ff1861a571455",
    "comment": "some more words",
    "meansOfAwareness": "awarenesss",
    "frequency": 12,
    "averageNumberOfBeneficiariesPerService": 3,
    "totalNumberOfBeneficiaries": 100,
    "serviceStatus": "Ongoing"
}
```
- to edit/update a beneficiary
`PUT /v1/stakeholders-directory/<id:beneficiary_id>/beneficiaries`
- to delete the beneficiary
`/v1/stakeholders-directory/<id:beneficiary_id>/beneficiaries`

**Any background context you want to provide?**

Any pertinent information that should be considered

**What are the relevant Jira issues?**

[MMDP-382](https://viisaus.atlassian.net/browse/MMDP-382)

**Screenshots (if appropriate)**
- creating a stakeholder
image

- update stakeholder
image

**Questions:**

### CODE REVIEW

File: server/models/stakeholdersDirectory/ReturneeService.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import keystone from 'keystone';
+
+const ReturneeService = new keystone.List('ReturneeService');
+const { Types } = keystone.Field;
+
+ReturneeService.add({
+  serviceName: { type: String, unique: true },
+  stakeholder_id: { type: Types.Relationship, ref: 'Stakeholder' },
```
Comment (p8ul) [inline server/models/stakeholdersDirectory/ReturneeService.js]: Please refactor this javascript object to use camelCase. You have a mixture of snakeCase, camelCase, PascalCase naming conventions. Please follow the team code conventions. Thank you

https://github.com/airbnb/javascript#naming--camelCase

https://github.com/airbnb/javascript#naming--PascalCase

Comment (fgkinus) [reply server/models/stakeholdersDirectory/ReturneeService.js]: Thanks, @p8ul for pointing this out. I have fixed the naming of variables accordingly

**Comment by p8ul** [comment]
Thanks you for following camelCase convention. You should also consider doing away with `status` object in your response. And if you must use then remove the typo `successes` should be replaced with `success`

Screenshot 2019-03-12 at 08 02 23

**Comment by fgkinus** [comment]
> Hello @fgkinus , thanks for the good work. I was proposing we create a folder insider `stakeholdersDirectory` called `dropDowns` where we place all dropdown models. I think this will make the `stakeholdersDirectory` more organised.

@zaabu sure. I think that makes sense, I will do that and get back to you

**Comment by fgkinus** [comment]
> Thanks you for following camelCase convention. You should also consider doing away with `status` object in your response. And if you must use then remove the typo `successes` should be replaced with `success`
> 
> Screenshot 2019-03-12 at 08 02 23

Resolved

**Pull request merged**